clients/drcachesim/tools/schedule_stats.cpp

```diff
@@ -225,6 +225,31 @@ schedule_stats_t::record_context_switch(per_shard_t *shard, int64_t workload_id,
             ++shard->counters.voluntary_switches;
         if (shard->direct_switch_target == tid)
             ++shard->counters.direct_switches;
+        if (shard->last_syscall_number >= 0) {
+            if (shard->stream->get_version() < TRACE_ENTRY_VERSION_FREQUENT_TIMESTAMPS) {
+                // Legacy versions do not have the timestamps to compute latencies.
+            } else {
+                assert(shard->pre_syscall_timestamp > 0);
```

Comment:
What happens if the trace starts at a syscall? I forget the nuances related to attaching when a thread is mid-syscall, and align-endpoints:

For !SA_RESTART, the syscall will return EINTR, will we have the syscall in the trace?

For SA_RESTART, I guess we will have the syscall in the trace, with the ~attach timestamp before the syscall?

Reply:
We will not have the TYPE_SYSCALL marker in that situation.  If it's recorded (-record_syscall) we will have FUNC_ID and FUNC_RETVALL and SYSCALL_FAILED.  I added two cases of such a thread to the test here: starting out on a core (3rd core in the test), and being switched to after a full syscall in another thread (after C's syscall on 2nd core).  For both I have a preempt switching right after that initial post-syscall.  The test still passes: confirms nothing is recorded for latencies for these post-syscall start points.